### ISSUE TITLE
Invalid Nullable Documentation

### ISSUE THREAD

**Issue by donatj** [scrutinizer-ci/php-analyzer#795]
I  receiving the following message on this and similar definitions.
> Should the type for parameter $linkingUserRoles not be null|integer[]?

```php
	/**		
	 * Gets the user ids for a specific building		
	 * @param  int   $buildingId       ID of the building to find		
	 * @param  int[] $linkingUserRoles list of roles to consider when looking up course ownership		
	 * @return int[]                   A list of user ids		
	 */		
	public function getCourseIdsFromBuildingId( $buildingId, array $linkingUserRoles = null  ) {
```

Notice the definition has ***array*** in it, `array $linkingUserRoles = null`, the user of the method cannot pass a null, the fact that it's null if you skip it is simply an implementation detail, not part of the API - so documenting it as such would be incorrect.

see https://scrutinizer-ci.com/g/capdig/myon/inspections/e0d39ad1-a82b-4f01-a6d4-39394a5090aa/issues/files/application/classes/Mappers/BuildingMapper.php?status=new&orderField=path&order=asc&issueId=76725657-e65a-4905-90e0-8ce87e65cddd

**Comment by nasimnabavi** [comment]
The default value should be array(), not null.
Best

**Comment by donatj** [comment]
There are many times you want the default behavior to vary from empty array and that is why php allows the default to be null on a type hinted field. That answer is wrong.
